src/services/identity/LaunchClient.ts

```diff
@@ -48,8 +70,107 @@ class LaunchClient {
 
   sendGetDomainAssociationCommand = (
     input: GetDomainAssociationCommandInput
-  ): Promise<GetDomainAssociationCommandOutput> =>
-    this.amplifyClient.send(new GetDomainAssociationCommand(input))
+  ): Promise<GetDomainAssociationCommandOutput> => {
+    if (this.shouldMockAmplifyDomainCalls()) {
+      // handle mock input
+      return this.mockGetDomainAssociationOutput(input)
+    }
+    return this.amplifyClient.send(new GetDomainAssociationCommand(input))
+  }
+
+  /**
+   * The rate limit for Create Domain Association is 10 per hour.
+   * We want to limit interference with operations, as such we mock this call during development.
+   * @returns Mocked output for CreateDomainAssociationCommand
+   */
+  mockCreateDomainAssociationOutput = (
+    input: CreateDomainAssociationCommandInput
+  ): Promise<CreateDomainAssociationCommandOutput> => {
+    // We are mocking the response from the Amplify API, so we need to store the input
+    this.mockDomainAssociations.set(input.domainName, input.subDomainSettings)
+    const mockResponse: CreateDomainAssociationCommandOutput = {
+      $metadata: {
+        httpStatusCode: 200,
+      },
+      domainAssociation: {
+        autoSubDomainCreationPatterns: [],
+        autoSubDomainIAMRole: undefined,
+        certificateVerificationDNSRecord: undefined,
+        domainAssociationArn: `arn:aws:amplify:ap-southeast-1:11111:apps/${input.appId}/domains/${input.domainName}`,
+        domainName: input.domainName,
+        domainStatus: "CREATING",
+        enableAutoSubDomain: false,
+        statusReason: undefined,
+        subDomains: undefined,
+      },
+    }
+
+    const subDomainSettingsList = input.subDomainSettings
+    if (!subDomainSettingsList || subDomainSettingsList.length === 0) {
+      return Promise.resolve(mockResponse)
+    }
+
+    const subDomains: SubDomain[] = this.getSubDomains(subDomainSettingsList)
+
+    // We know that domainAssociation is not undefined, so we can use the non-null assertion operator
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    mockResponse.domainAssociation!.subDomains = subDomains
+    return Promise.resolve(mockResponse)
+  }
+
+  private shouldMockAmplifyDomainCalls(): boolean {
+    return config.get("aws.amplify.mockAmplifyCreateDomainAssociationCalls")
+  }
+
+  private getSubDomains(
+    subDomainList: SubDomainSetting[],
+    isCreated = false
+  ): SubDomain[] {
+    const subDomains = subDomainList
+      .map((subDomain) => subDomain.prefix)
+      .filter((prefix) => prefix !== undefined && prefix !== null)
+      .map((subDomainPrefix) => ({
+        dnsRecord: `${subDomainPrefix} CNAME ${
+          isCreated ? "test.cloudfront.net" : "<pending>"
+        }`,
+        subDomainSetting: {
+          branchName: "master",
+          prefix: subDomainPrefix,
+        },
+        verified: false,
+      })) as SubDomain[]
```

Comment:
why do we need the cast here?

Reply:
Filter doesn't narrow types :( 
Primarily this stems from `subDomainPrefix` not being narrowed to `string` after the filer, I have [changed](602b6f8bc03138ee091d291b2d54864d262f466c) the code to reflect this. 

Am opting against using a type guard here since this only checks if a string is undefined, opting for the more simplistic solution here